cmd/readmem/src/lib.rs

```diff
@@ -199,6 +205,24 @@ fn readmem(context: &mut ExecutionContext) -> Result<()> {
         bail!("address must be {}-byte aligned", size);
     }
 
+    if let Some(file) = subargs.file {
+        let mut f = std::fs::File::create(&file)?;
+        for (i, addr) in (addr..addr + (length as u32)).step_by(max).enumerate()
+        {
+            let mut bytes = vec![
+                0u8;
+                std::cmp::min(
+                    max,
+                    length.checked_sub(i * max).unwrap_or(length)
```

Comment:
Can this ever actually underflow?  I can't think of a way that `i * max` would ever exceed `length`.

Reply:
I think you're right and I was being too cautious here